src/Lib/Managers/ExportAreaManager.vala

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) 2020 Alecaddd (https://alecaddd.com)
+ *
+ * This file is part of Akira.
+ *
+ * Akira is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * Akira is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Alessandro "Alecaddd" Castellani <castellani.ale@gmail.com>
+ */
+
+public class Akira.Lib.Managers.ExportAreaManager : Object {
+    private const string COLOR = "#41c9fd";
+    private const double LINE_WIDTH = 2.0;
+    private const double MIN_SIZE = 1.0;
+
+    public weak Akira.Lib.Canvas canvas { get; construct; }
+    public Akira.Dialogs.ExportDialog export_dialog;
+
+    private double initial_x;
+    private double initial_y;
+    private double initial_width;
+    private double initial_height;
+
+    public Goo.CanvasRect area;
+    public Cairo.Format format;
+    public Cairo.Surface surface;
+    public Cairo.Context context;
+    public Gdk.PixbufLoader loader;
+    public Gdk.Pixbuf pixbuf;
+
+    public ExportAreaManager (Akira.Lib.Canvas canvas) {
+        Object (
+            canvas: canvas
+        );
+    }
+
+    public Goo.CanvasRect create_area (Gdk.EventButton event) {
+        var dash = new Goo.CanvasLineDash (2, 5.0, 5.0);
```

Comment:
We need to account for the zoom here, so the `CanvasLineDash` line width needs to be divided by the current scale otherwise, at higher zoom, the line width is too high.

Reply:
Good point, I completely forgot.